massa-models/src/operation.rs

```diff
@@ -24,6 +28,14 @@ const OPERATION_ID_STRING_PREFIX: &str = "OPE";
 #[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
 pub struct OperationId(Hash);
 
+/// Left part of the operation id hash stored in a vector of size [OPERATION_ID_PREFIX_SIZE_BYTES]
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct OperationPrefixId(Vec<u8>);
+
+/// Right part of the operation id hash, contains the remains of `OperationId - OperationPrefixId`
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct OperationSuffixId(Vec<u8>);
```

Comment:
I let a comment for that part. My guess is that we could check if the performance is better if we store a prefix-suffix map instead of a prefix-id map. I'll move that part in another branch.